src/protocol.h

```diff
@@ -103,7 +103,8 @@ class CAddress : public CService
             Init();
         if (nType & SER_DISK)
             READWRITE(nVersion);
-        if ((nType & SER_DISK) || !(nType & SER_GETHASH))
+        if ((nType & SER_DISK) ||
+            (nVersion >= CADDR_TIME_VERSION && !(nType & SER_GETHASH)))
```

Comment:
This reversion is also necessary. ``nVersion`` here can refer either to ``INIT_PROTO_VERSION`` (when deserializing the address fields in the initial "version" message received from a peer), or to the final negotiated version (when deserializing later messages).